code/game/objects/items/devices/radio/radio.dm

```diff
@@ -306,7 +306,7 @@
 
 /obj/item/radio/Hear(message, atom/movable/speaker, message_language, raw_message, radio_freq, list/spans, list/message_mods = list())
 	. = ..()
-	if(radio_freq || !broadcasting || get_dist(src, speaker) > canhear_range)
+	if(radio_freq || !on || !broadcasting || get_dist(src, speaker) > canhear_range)
```

Comment:
Do these stop working if EMPed?

Reply:
Yes they do.

Reply:
@PowerfulBacon 
![image](https://user-images.githubusercontent.com/9547572/170862873-b39e0ed7-c745-4e15-acc6-d8801653418b.png)

Works this way for all radios and headset radios.